plugins/afr/validate.go

```diff
@@ -9,12 +9,39 @@ import (
 
 var names = [...]string{"replicate", "afr"}
 
+var shdEnabledKey = [...]string{"afr.self-heal-daemon", "replicate.self-heal-daemon"}
+
+// IsVolReplicate return true if volume is of type replicate, disperse, distreplicate or distdisperse
+// otherwise it returns false
+func IsVolReplicate(vType volume.VolType) bool {
+	if vType == volume.Replicate || vType == volume.Disperse || vType == volume.DistReplicate || vType == volume.DistDisperse {
+		return true
+	}
+
+	return false
+}
+
+// IsHealEnabled returns true if heal is enabled for the volume otherwise returns false.
+func IsHealEnabled(v *volume.Volinfo) bool {
```

Comment:
For now, duplicate (and unexport) this code in each package where it's used. In future we'll merge afr and selfheal packages.